docs/data/date-pickers/date-field/date-field.md

```diff
@@ -32,11 +32,14 @@ Learn more about the _Controlled and uncontrolled_ pattern in the [React documen
 
 {{"demo": "CustomDateFormat.js"}}
 
+:::info
```

Comment:
IMHO this demo should either not exist or be inside the "Localization" section below
THis would help people understand that the custom date format is part of the the Localization in our doc

Reply:
Initially, I was leaning towards that direction, but then I thought that the various custom formats for different components (date, time, and date-time) have lots of examples. If we want to provide at least a similar amount of demos in the `Custom format` section, it would be huge. 🤯 
That's why I decided to keep the separate section. 🤔 
WDYT?
Would moving those `Customize the <x> format` sections as sub-sections of `Localization` make more sense to you?

Reply:
@flaviendelangle I've moved the `Customize the format` sections as subsections of `Localization`.
Let me know what you think. 😉

Reply:
I can be a good trade-off :ok_hand:

Reply:
This can be a great use-case for a switch like github does it on their webhook documentation:
<img width="801" alt="Screenshot 2024-02-08 at 14 20 15" src="https://github.com/mui/mui-x/assets/32863416/aaec8f8c-413e-457a-bbec-f3cc3ca3eae4">

Reply:
In the sense that you could provide a few options to select formats and then the fields will use that. WDYT?

Reply:
That is a great suggestion, I'll explore that option. 👌